firefox-ios/firefox-ios-tests/Tests/ClientTests/SummarizerServiceTests.swift

```diff
@@ -0,0 +1,110 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+@testable import Client
+import XCTest
+import WebKit
+
+final class SummarizerServiceTests: XCTestCase {
+    static let mockResponse = ["Summarized", "content"]
+    static let maxWords = 100
+    static let maxWordCount = 50
+    var mockWebView = WKWebViewMock(URL(string: "https://foo.com")!)
+
+    func testSummarizerServiceReturnsSummary() async throws {
+        let subject = createSubject()
+        let result = try await subject.summarize(from: mockWebView)
+        XCTAssertEqual(result, Self.mockResponse.joined(separator: " "))
+    }
+
+    func testSummarizerServiceThrowsWhenContentTooLong() async {
+        let checker = MockSummarizationChecker(
+            canSummarize: false,
+            reason: .contentTooLong,
+            wordCount: Self.maxWordCount,
+            textContent: nil
+        )
+
+        let subject = createSubject(checker: checker)
+
+        await assertSummarizeThrows(.tooLong) {
+            _ = try await subject.summarize(from: self.mockWebView)
+        }
+    }
+
+    func testSummarizerServiceReturnsStreamedSummary() async throws {
+        let subject = createSubject()
+        var streamedChunks: [String] = []
+
+        for try await chunk in subject.summarizeStreamed(from: mockWebView) {
+            streamedChunks.append(chunk)
+        }
+
+        XCTAssertEqual(streamedChunks, Self.mockResponse)
```

Comment:
similar to above, its easier to make the distinction here without having to scroll back up to see `mockResponse`
```suggestion
        XCTAssertEqual(streamedChunks, ["Summarized", "content"] )
```